### ISSUE TITLE
 Can't run tests without manual intervention after a reboot when security is set

### ISSUE THREAD

**Issue by nextlevelbeard** [appium/appium#15681]
## Reproduction steps
1. Successfully run tests on an emulator (Appium Settings is installed/runnning)
2. Set a pin lock
3. Reboot it (device is locked)
5. Try to run tests with capabilities that have `unlockType` and `unlockKey`, it fails

To unlock, Appium Settings service needs to be running, which is not.
You can't start the service manually without unlocking the device first either.

* Appium version (or git revision) that exhibits the issue: latest

**Comment by mykola-mokhnach** [comment]
I don't think that Appium ever supported such testing scenarios. Feel free to push a PR that implements the support for it

**Comment by nextlevelbeard** [comment]
@mykola-mokhnach 
- Do you know if the same unlocking procedure that is done currently relying on `unlockType` and `unlockType` can be used here?
- Does it have a dependency on Appium Settings?
- Would calling the `unlock` command work here?
- Is it true that to unlock, Appium Settings service needs to be running in the first place?

**Comment by mykola-mokhnach** [comment]
No, unlock on its own does not rely on Appium Settings. Although, being a session capability it requires all other driver prerequisites to start and Settings app is one of these prerequisites. You could learn this logic in more details from https://github.com/appium/appium-android-driver/blob/master/lib/android-helpers.js
